settings/src/script.js

```diff
@@ -61,20 +61,32 @@ function Main( { userId } ) {
 		hasPrimaryProvider,
 	} = user;
 	const [ globalNotice, setGlobalNotice ] = useState( '' );
+	const totpEnabled = record?.[ '2fa_available_providers' ].includes( 'Two_Factor_Totp' );
+	const backupCodesEnabled =
+		record?.[ '2fa_available_providers' ].includes( 'Two_Factor_Backup_Codes' );
+	const webAuthnEnabled = record?.[ '2fa_available_providers' ].includes(
+		'TwoFactor_Provider_WebAuthn'
+	);
```

Comment:
I feel like passing these props down rather than getting from the user context is a backward step. Maybe we should derive them there, similar to `hasPrimaryProvider`, and we can also add tests to make sure they handle the record being undefined.

utilities.js `useUser` would look something like:

```
const availableProviders = userRecord.record?.[ '2fa_available_providers' ] ?? [];

return {
	userRecord: { ...userRecord },
	isSaving,
	hasTotp: availableProviders.includes(  'Two_Factor_Totp'  ),
	hasBackupCodes: availableProviders.includes(  'Two_Factor_Backup_Codes'  ),
	hasWebAuthn = availableProviders.includes( 'TwoFactor_Provider_WebAuthn'	),
	hasPrimaryProvider: hasTotp || hasWebAuthn,
};
```

Reply:
That's a good idea. When referring to passing down these variables via `Context` in the first place, I was initially thinking of passing them down [here](https://github.com/WordPress/wporg-two-factor/blob/trunk/settings/src/script.js#L192), so I didn't think it would make things better. But placing it in `useUser` is a good choice! Thanks.

Reply:
Commit pushed: [3c36cd9](https://github.com/WordPress/wporg-two-factor/pull/197/commits/3c36cd9b004ff706cd02a238bacb4f3342dc4399)